abi.xml

```diff
@@ -0,0 +1,1655 @@
+    <elf-symbol name='safe_mkdir_p_dir' type='func-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='systemd_escape' type='func-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='tunnel_mode_to_string' type='func-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='unlink_glob' type='func-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='wifi_get_freq24' type='func-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='wifi_get_freq5' type='func-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='write_netplan_conf' type='func-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='write_netplan_conf_full' type='func-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+  </elf-function-symbols>
+  <elf-variable-symbols>
+    <elf-symbol name='NETPLAN_OPTIONAL_ADDRESS_TYPES' size='96' type='object-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='NETPLAN_WIFI_WOWLAN_TYPES' size='160' type='object-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='global_backend' type='object-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='global_state' size='168' type='object-type' binding='global-binding' visibility='default-visibility' alias='netdefs' is-defined='yes'/>
+    <elf-symbol name='netdefs' type='object-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='netdefs_ordered' type='object-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='netplan_backend_to_name' size='32' type='object-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='ovs_settings_global' type='object-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='wifi_frequency_24' size='8' type='object-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+    <elf-symbol name='wifi_frequency_5' size='8' type='object-type' binding='global-binding' visibility='default-visibility' is-defined='yes'/>
+  </elf-variable-symbols>
+  <abi-instr version='1.0' address-size='64' path='src/abi_compat.c' comp-dir-path='/root/netplan' language='LANG_C99'>
+    <class-decl name='netplan_state' size-in-bits='1344' is-struct='yes' visibility='default' filepath='src/types.h' line='433' column='1' id='type-id-1'>
```

Comment:
The XML contains everything, but `abidiff` will only consider the public symbols (as found via the `--headers-dir2 include/` argument). So it will not complain about shuffling members of `netplan_state`:
```
$ abidiff abi-comapt/impish_0.104.xml libnetplan.so.0.0 --headers-dir2 include/
Functions changes summary: 0 Removed, 0 Changed (33 filtered out), 0 Added functions
Variables changes summary: 0 Removed, 0 Changed, 0 Added variable
```

But will fail on it if we define `types.h` as public API (via `--header-file2 src/types.h`):
```
$ abidiff abi-comapt/impish_0.104.xml libnetplan.so.0.0 --headers-dir2 include/ --header-file2 src/types.h 
Functions changes summary: 0 Removed, 1 Changed (32 filtered out), 0 Added functions
Variables changes summary: 0 Removed, 0 Changed, 0 Added variable

1 function with some indirect sub-type change:

  [C] 'function netdef_pertype_iter* _netplan_state_new_netdef_pertype_iter(NetplanState*, const char*)' at abi_compat.c:214:1 has some indirect sub-type changes:
    parameter 1 of type 'NetplanState*' has sub-type changes:
      in pointed to type 'typedef NetplanState' at netplan.h:29:1:
        underlying type 'struct netplan_state' at types.h:433:1 changed:
          type size hasn't changed
          4 data member changes:
            'GHashTable* netplan_state::netdefs' offset changed from 0 to 1152 (in bits) (by +1152 bits)
            'GList* netplan_state::netdefs_ordered' offset changed from 64 to 1216 (in bits) (by +1152 bits)
            'NetplanBackend netplan_state::backend' offset changed from 128 to 1280 (in bits) (by +1152 bits)
            'NetplanOVSSettings netplan_state::ovs_settings' offset changed from 192 to 0 (in bits) (by -192 bits)
```

So this works as intended for `netplan_state` and others. But might actually be a problem for `struct netplan_net_definition`, as that is not considered public, but we want to know about breakage, for abi_compat reasons. So we might need to move `netplan_net_definition` into it's own (private) header file, which we pass on to `abidiff`, to make it consider this struct as public for the check.